### PR TITLE
binomial and exponential distributions, update on base to integrate floats (?)

### PR DESCRIPTION
**Pull request by Jules-Deschamps**

### CODE REVIEW

File: tests/data/binomial_data.py [hunk 1]
```diff
@@ -0,0 +1,147 @@
...
+    shape_list = [(1,)]
+    n_points_list = random.sample(range(5), 2)
+    n_vecs_list = random.sample(range(2, 5), 1)
+
+    def belongs_test_data(self):
+        smoke_data = [
+            dict(n_draws=10, vec=0.1, expected=True),
+            dict(n_draws=8, vec=-0.8, expected=False),
+            dict(n_draws=5, vec=0.1, expected=True),
+            dict(n_draws=2, vec=-1.0, expected=False),
```
Comment (alebrigant) [inline tests/data/binomial_data.py:22]: - `vec` is misleading as your are testing a point on the manifold and not a tangent vector. 
- Values higher than 1 are not tested.
- For one point, maybe test `gs.array([x])` as well as float `x`.
- Several points are not tested, i.e. `gs.array([x, y])` as well as `[x, y]` and `gs.array([[x],[y]])`.

File: tests/data/exponential_data.py [hunk 2]
```diff
@@ -0,0 +1,137 @@
...
+    n_points_list = random.sample(range(5), 2)
+    n_vecs_list = random.sample(range(2, 5), 1)
+
+    def belongs_test_data(self):
+        smoke_data = [
+            dict(vec=0.1, expected=True),
+            dict(vec=-0.8, expected=False),
+            dict(vec=0.1, expected=True),
+            dict(vec=-1.0, expected=False),
+        ]
```
Comment (alebrigant) [inline tests/data/exponential_data.py:23]: Similar comments as for binomial.

**Review by alebrigant — CHANGES_REQUESTED** [review]
Great ! Just some last minor comments to address and we can merge.

File: geomstats/geometry/base.py [hunk 3]
```diff
@@ -48,11 +48,15 @@ def belongs(self, point, atol=gs.atol):
         belongs : array-like, shape=[...,]
             Boolean evaluating if point belongs to the space.
         """
+        point = gs.array(point)
         minimal_ndim = len(self.shape)
+        if self.shape[0] == 1 and len(point.shape) <= 1:
```
[the remark this answers is not in the document]
Comment (Jules-Deschamps) [reply geomstats/geometry/base.py:53]: if is input an array that is already in the wanted form (eg for the dimension 1 exception we are dealing with, gs.array([[0],[1]]), deleting this condition would eventually return gs.array([0,1]) which is not what we want.
for all types of possible inputs in dimension 1 (ie float, gs.array(float), gs.array([float]), gs.array([float1, float2]), gs.array([[float1],[float2]])), belongs will "try its conditions" on gs.array([[float]]) for a single vector and for gs.array([[float1], [float2]]) for 2 vectors.

File: geomstats/information_geometry/exponential.py [hunk 4]
```diff
@@ -0,0 +1,177 @@
...
+from geomstats.geometry.base import OpenSet
+from geomstats.geometry.euclidean import Euclidean
+from geomstats.geometry.riemannian_metric import RiemannianMetric
+
+
+class ExponentialDistributions(OpenSet):
+    """Class for the manifold of exponential distributions.
+
+    This is the parameter space of binomial distributions with fixed number of draws
+    i.e. the unit segment.
```
[the remark this answers is not in the document]
Comment (Jules-Deschamps) [reply geomstats/information_geometry/exponential.py:19]: yes there have been some confusions between the two i will take care of it

File: geomstats/information_geometry/binomial.py [hunk 5]
```diff
@@ -0,0 +1,196 @@
...
+from geomstats.geometry.euclidean import Euclidean
+from geomstats.geometry.riemannian_metric import RiemannianMetric
+
+
+class BinomialDistributions(OpenSet):
+    """Class for the manifold of binomial distributions.
+
+    This is the parameter space of binomial distributions
+    i.e. the half-line of positive reals.
+    """
```
Comment (alebrigant) [inline geomstats/information_geometry/binomial.py:20]: Remaining inversion in the docstrings : positive reals <-> unit interval.

File: geomstats/information_geometry/binomial.py [hunk 6]
```diff
@@ -0,0 +1,196 @@
...
+
+
+class BinomialFisherRaoMetric(RiemannianMetric):
+    """Class for the Fisher information metric on binomial distributions.
+
+    References
+    ----------
+    Atkinson, C., & Mitchell, A. F. (1981). Rao's distance measure.
+    Sankhyā: The Indian Journal of Statistics, Series A, 345-365.
+    """
```
Comment (alebrigant) [inline geomstats/information_geometry/binomial.py:171]: .. [AM1981] Atkinson, C., & Mitchell, A. F. (1981). Rao's distance measure.
    Sankhyā: The Indian Journal of Statistics, Series A, 345-365.

File: geomstats/information_geometry/exponential.py (hunk 4 above)
[the remark this answers is not in the document]
Comment (alebrigant) [reply geomstats/information_geometry/exponential.py:19]: Remaining inversion unit segment <-> positive reals.

File: geomstats/information_geometry/exponential.py [hunk 7]
```diff
@@ -0,0 +1,177 @@
...
+            ]
+
+            return pdf_at_x
+
+        return pdf
+
+
+class ExponentialFisherRaoMetric(RiemannianMetric):
+    """Class for the Fisher information metric on exponential distributions."""
+
```
[the remark this answers is not in the document]
Comment (alebrigant) [reply geomstats/information_geometry/exponential.py:164]: .. [AM1981] Atkinson, C., & Mitchell, A. F. (1981). Rao's distance measure.
Sankhyā: The Indian Journal of Statistics, Series A, 345-365.

File: tests/data/binomial_data.py (hunk 1 above)
Comment (alebrigant) [reply tests/data/binomial_data.py:22]: `vec` -> `point`

**Comment by Jules-Deschamps** [comment]
doesn't work with new "triangle_dist" but close because of history issues

**Pull request closed without merging**